infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -186,13 +234,13 @@ public void eventReceived(Action action, Pod pod) {
                 public void onClose(KubernetesClientException cause) {
                   future.completeExceptionally(
                       new InfrastructureException(
-                          "Waiting for pod '" + name + "' was interrupted"));
+                          "Waiting for pod '" + podName + "' was interrupted"));
                 }
               });
 
       Pod actualPod = podResource.get();
       if (actualPod == null) {
-        throw new InfrastructureException("Specified pod " + name + " doesn't exist");
+        throw new InfrastructureException("Specified pod " + podName + " doesn't exist");
```

Comment:
There is a situation when deployment was specified and the message may be confusing because it says that podName was specified while it wasn't.

Reply:
Yeah this is a tricky one. One of the things I'm not quite happy with in this PR is that as a result of trying to not affect the rest of the internal workspaces representation (as Pods are used extensively), I've had to keep the underlying way of running pods invisible to che-server. While this allows most of Che to work without any modification, within this class, it makes it confusing as you point out, as it's possible to call this method with a deployment name. 

If this message were to be logged, the two causes are 
1. The method is called with a deployment name, and `getPodName()` succeeds, returning the name of a running pod controlled by the specified deployment, *and* that running pod terminates between the call to `getPodName()` and this `get()` call. In this case, the log message should be something along the lines of `No pods in Deployment [name] found`
2. The method is called with a pod name directly (i.e. there is no deployment), and the pod that we `get()` in the `getPodName()` method terminates before we reach this point. In this case ideally we would log the message currently there. 

Does this make sense, or can we make it more understandable? This check should really never fail, as `getPodName` always calls `get()` on the same `PodResource` and throws an error earlier if `null` is encountered.

Reply:
I've modified the logging to reflect the above, leveraging the fact that if `name.equals(podName)` then `name` refers to a bare pod. It does make the already-long error handling block in these methods longer, however. Let me know what you think.